### PR TITLE
Chaosblade operator

### PR DESCRIPTION
**Pull request by oeular**
### Checklist

I have:

- [x] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [x] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [x] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [x] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

### Verified Addon Contribution

I have:

- A new addon added in this repo should be put in as an experimental one unless you have test for a long time in your product environment and be approved by most maintainers.

- An experimental addon must meet these conditions to be promoted as a verified one.

  - [x] This addon must be tested by addon's [e2e-test](./test/e2e-test/addon-test) to guarantee this addon can be enabled successfully.

  - This addon must have some basic but necessary information.

    - [x] An accessible icon url and source url defined in addon's `metadata.yaml`.
    
    - [x] A detail introduction include a basic example about how to use and what's the benefit of this addon in `README.md`.
      
    - [x] Also provide an introduction in KubeVela [documentation](https://kubevela.net/docs/reference/addons/overview).
    
    - [x] It's more likely to be accepted if useful examples provided in example [dir](examples/).

### CODE REVIEW

File: experimental/addons/chaosblade-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    action: delete
+    desc: "delete pod by names"
+    matchers:
+    - name: names
+      value:
+      - "kubevela-vela-core-8699fb6d68-mrfvv"
+    - name: namespace
+      value:
+      - "vela-system"
+EOF
```
Comment (wonderflow) [inline experimental/addons/chaosblade-operator/README.md:66]: can it be a trait, when this trait added, than the app( with pod detected) will automatically be injected a chaos and get the result integrating with the Grafana addon.

File: experimental/addons/chaosblade-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      - expose: true
+        port: 80
+        protocol: TCP
+        ports: 80
+    traits:
+    - type: delete-pod-by-names
+      properties:
+        bladeName: delete-po-chaos
+        podName:
+        - nginx-demo-6cbfdc95df-q7vmn
```
Comment (wonderflow) [inline experimental/addons/chaosblade-operator/README.md:103]: how does it work? every pod has unique name when deploy.

File: experimental/addons/chaosblade-operator/README.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        port: 80
+        protocol: TCP
+        ports: 80
+EOF
+```
+
+Get the pod name to make a chaosblade
+```shell
+$ kubectl get po -n default | grep nginx
+nginx-demo-6cbfdc95df-q7vmn   1/1     Running   0                 33s
```
Comment (wonderflow) [inline experimental/addons/chaosblade-operator/README.md:76]: I think this is a misleading as vela users shouldn't get into pods to learn how to write applications

File: experimental/addons/chaosblade-operator/template.cue [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                                        metadata: name: parameter.namespace
+                                }]
+                        },
+                        {
+                            name: "blade-helm"
+                            type: "helm"
+                            dependsOn: ["blade-ns"]
+                            properties: {
+                                    repoType: "helm"
+                                    url:      "https://oeular.github.io/chaosblade-operator/"
```
Comment (wangyikewxgm) [inline experimental/addons/chaosblade-operator/template.cue:24]: Can we ref to an official address?

File: experimental/addons/chaosblade-operator/README.md (hunk 2 above)
Comment (oeular) [reply experimental/addons/chaosblade-operator/README.md:103]: This trait is to delete the pod by name so as to make chaos.

File: experimental/addons/chaosblade-operator/README.md (hunk 3 above)
Comment (oeular) [reply experimental/addons/chaosblade-operator/README.md:76]: In this demo, it is necessary to know the pod name.

File: experimental/addons/chaosblade-operator/template.cue (hunk 4 above)
Comment (oeular) [reply experimental/addons/chaosblade-operator/template.cue:24]: The official helm chart repo is not ready. I have made an issue for them.

**Review by wonderflow — DISMISSED** [review]
LGTM, please trime the mongodb data

File: experimental/addons/chaosblade-operator/definitions/fail-pod-by-labels.cue [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		podDisruptive:   false
+		workloadRefPath: ""
+	}
+	description: ""
+	labels: {}
+	type: "trait"
+}
+
+template: {
+	output: {
```
Comment (wangyikewxgm) [inline experimental/addons/chaosblade-operator/definitions/fail-pod-by-labels.cue:17]: I‘m not sure is a traitDefinition can use `output` ? trait must use `outputs`.

**Review by wangyikewxgm — APPROVED** [review]
LGTM,thanks!

**Pull request merged**